staging/mausezahn.c

```diff
@@ -710,11 +719,20 @@ int getopts (int argc, char *argv[])
 				(ipv6_mode && get_ip6_range_dst(tx.ip_dst_txt, l)) || // returns 1 when no range has been specified
 				(!ipv6_mode && get_ip_range_dst(tx.ip_dst_txt)))
 			{
-				// name2addr4 accepts a DOTTED DECIMAL ADDRESS or a FQDN:
-				if (ipv6_mode)
+				// name2addr{4, 6} accept a valid IP address or a FQDN:
+				if (ipv6_mode) {
 					tx.ip6_dst = libnet_name2addr6 (l, tx.ip_dst_txt, LIBNET_RESOLVE);
-				else
-					tx.ip_dst = libnet_name2addr4 (l, tx.ip_dst_txt, LIBNET_RESOLVE);
+					if (strncmp((char*)&tx.ip6_dst,(char*)&in6addr_error,sizeof(in6addr_error))==0) {
+					  fprintf(stderr, "Failed to set destination IPv6 address. Please check if destination is set to a valid IPv6 address.\n");
+					  return 1;
+					}
+				}
+				else {
+					if ((tx.ip_dst = libnet_name2addr4 (l, tx.ip_dst_txt, LIBNET_RESOLVE)) == -1){
+					  fprintf(stderr, "Failed to set destination IPv4 address. Please check if destination is set to a valid IPv4 address.\n");
+					  return 1;
+					}
+			  }
 			}
```

Comment:
I used a combination of both methods; the `libnet_name2addr` functions for all but the broadcast addresses, and `inet_pton` for covering those edge cases. This covers the additional cases pointed out by @vkochan and I have updated my [test cases](https://gist.github.com/mandarg/0f7d6d6a09f378f5d74f94757c077d4c) to include those.

I  have tried to make things as consistent as possible with the rest of the code in mausezahn.c, although as mentioned, it has inconsistencies itself in several places. (On a related note, would there be any interest in auto-formatting this using something like `astyle`? `git blame` etc. would still work as expected with `-w`.)